lib/Ix/DBIC/Result.pm

```diff
@@ -136,6 +167,91 @@ sub ix_add_properties ($class, @pairs) {
   return;
 }
 
+sub add_array_type ($class, $prop, $def) {
+  my $type = $def->{data_type};
+
+  my $ix_type = $IX_TYPE{ $type };
+
+  Carp::confess("Attempt to add property $prop with unknown data_type $type")
+    unless $ix_type && $ix_type->{data_type};
+
+  my $data_type = $ix_type->{db_data_type} // $ix_type->{data_type};
+
+  my $belongs_to = $class->table;
+  $belongs_to =~ s/s$//;
+
+  my $table = $belongs_to . '_ix_at_' . $prop;
+
+  my $class_name = $class;
+  $class_name .= "::Ix::At::\u$prop";
+
+  my $pkg = <<"EOF";
+package $class_name;
+
+use strict;
+use warnings;
+
+use base qw/DBIx::Class::Core/;
+
+__PACKAGE__->table("$table");
+
+__PACKAGE__->add_columns(
+  id        => { data_type => 'bigserial' },
+  accountId => { data_type => 'integer' },
+  ${belongs_to}Id => { data_type => 'integer' },
+  created   => { data_type => 'timestamptz', default_value => \\'NOW()' },
+  value     => { data_type => '$data_type' },
+);
+
+__PACKAGE__->set_primary_key('id');
+
+__PACKAGE__->belongs_to(
+  $belongs_to => '$class',
+  {
+    'foreign.id' => 'self.${belongs_to}Id',
+    'foreign.accountId' => 'self.accountId',
+  },
+);
+
+# XXX
+#   my (\$self, \$sqlt_table) = \@_;
+
+#   \$sqlt_table->add_index(name => 'idx_name', fields => ['name']);
+# }
+EOF
+
+  eval $pkg;
+  die "Failed to eval $pkg: $@\n" if $@;
+
+  no strict 'refs';
+  *{"${class}::array_class"} = sub { return { $table => $class_name } };
+  $class->has_many(
+    $prop => $class_name,
+    {
+      "foreign.${belongs_to}Id" => "self.id",
+      "foreign.accountId"       => "self.accountId",
+    },
+  );
+
+  $def->{is_array_type} = 1;
+  $def->{array_class} = $class_name;
+  $def->{rs} = $table;
```

Comment:
This name confuses me through its vagueness.  Also, weird?  I'd expect the rs name to be more like the class than table name?  I'm just confused.